chainforge/react-server/src/backend/query.ts

```diff
@@ -291,8 +298,10 @@ export class PromptPipeline {
     let query: Dict | undefined;
     let response: Dict | LLMResponseError;
     try {
+      if (QueryTracker.has(id))  { throw new Error("Node cancelled"); }
```

Comment:
These errors should be a specific class you defined, like we did before for DuplicateVar --something like `UserForcedPrematureExit` --then here you throw that error, and you can later check for that error type using `instanceof` (https://javascript.info/custom-errors)